package.json

```diff
@@ -45,9 +47,9 @@
     "deploy-docs": "cd docs && yarn install && yarn deploy",
     "dev": "start-storybook -p 6006",
     "generate-icons": "./tools/generate-icons.js",
-    "lint": "eslint \"src/**/*.js\" --cache",
+    "lint": "eslint \"src/**/*.{js,ts,tsx}\" --cache",
     "postpublish": "branch=$(git rev-parse --symbolic-full-name --abbrev-ref HEAD); if [[ $branch == \"master\" ]]; then yarn update-docs || yarn deploy-docs; else echo \"not on main branch, skipping docs deploy\"; fi;",
-    "prepublishOnly": "rm -rf esm commonjs umd && yarn build",
+    "prepublishOnly": "rm -rf esm commonjs umd types && yarn build",
     "release": "np",
     "size": "size-limit",
```

Comment:
may want to add a tsc typecheck command and add it as an initial step in the build